src/toil/wdl/versions/v1.py

```diff
@@ -33,4 +48,736 @@ def analyze(self):
         intermediate data structures: `self.workflows_dictionary` and
         `self.tasks_dictionary`.
         """
-        raise NotImplementedError
+        lexer = WdlV1Lexer(FileStream(self.wdl_file))
+        parser = WdlV1Parser(input=CommonTokenStream(lexer))
+        tree = parser.document()
+        self.visitDocument(tree)
+
+    def visitDocument(self, ctx):
+        """
+        Root of tree. Contains `version` followed by an optional workflow and
+        any number of `document_element`s.
+        """
+        wf = ctx.workflow()
+        if wf:
+            self.visitWorkflow(wf)
+
+        for element in ctx.document_element():
+            self.visitDocument_element(element)
+
+    def visitDocument_element(self, ctx):
+        """
+        Contains one of the following: 'import_doc', 'struct', or 'task'.
+        """
+        element = ctx.children[0]
+        # task
+        if is_context(element, 'TaskContext'):
```

Comment:
I was thinking about subclassing this class for the development version, so we can reuse a lot of the code. 

Using `isinstance` would prevent us from doing that, because antlr4 will generate different parsers and classes (e.g.: `WdlV1Parser.TaskContext` and `WdlParser.TaskContext`) based on the grammar file. 

Please let me know if there is a better way!

Reply:
I think that makes sense. We might want to explain it in a docstring or comment, though, instead of just in a discussion deep in the Github comment history.